t/batches.t

```diff
@@ -0,0 +1,335 @@
+use strict;
+use warnings;
+use 5.14.2;
+
+use Cwd;
+use Data::Dumper;
+use File::Temp qw[tempdir];
+use JSON::PP;
+use POSIX qw( strftime );
+use Time::Local qw( timelocal_modern );
+use Test::Exception;
+use Test::More;    # see done_testing()
+
+use Zonemaster::Engine;
+
+# Use the TARGET environment variable to set the database to use
+# default to SQLite
+my $db_backend = $ENV{TARGET};
+if ( not $db_backend ) {
+    $db_backend = 'SQLite';
+} elsif ( $db_backend !~ /^(?:SQLite|MySQL|PostgreSQL)$/ ) {
+    BAIL_OUT( "Unsupported database backend: $db_backend" );
+}
+
+diag "database: $db_backend";
+
+my $tempdir = tempdir( CLEANUP => 1 );
+
+my $datafile = q{t/batches.data};
+if ( not $ENV{ZONEMASTER_RECORD} ) {
+    die q{Stored data file missing} if not -r $datafile;
+    Zonemaster::Engine->preload_cache( $datafile );
+    Zonemaster::Engine->profile->set( q{no_network}, 1 );
+    diag "not recording";
+} else {
+    diag "recording";
+}
+
+# Require Zonemaster::Backend::RPCAPI.pm test
+use_ok( 'Zonemaster::Backend::RPCAPI' );
+
+use_ok( 'Zonemaster::Backend::Config' );
+
+my $cwd = cwd();
+
+my $config = <<EOF;
+[DB]
+engine = $db_backend
+
+[MYSQL]
+host     = localhost
+user     = travis_zm
+password = travis_zonemaster
+database = travis_zonemaster
+
+[POSTGRESQL]
+host     = localhost
+user     = travis_zonemaster
+password = travis_zonemaster
+database = travis_zonemaster
+
+[SQLITE]
+database_file = $tempdir/zonemaster.sqlite
+
+[LANGUAGE]
+locale = en_US
+
+[PUBLIC PROFILES]
+test_profile=$cwd/t/test_profile.json
+EOF
+
+my $user = {
+    username => 'user',
+    api_key => 'key'
+};
+
+# define the default properties for the tests
+my $params = {
+    client_id      => 'Unit Test',
+    client_version => '1.0',
+    ipv4           => JSON::PP::true,
+    ipv6           => JSON::PP::true,
```

Comment:
I might be of value to be able to run with at least IPv6 turned off with an environment variable.

Reply:
Good idea (and same for IPv4). However this is outside the scope of this PR.

Reply:
It depends on how this test will work in a non-IPv6 environment. It will always be run when updated Zonemaster-Backend is installed, won't it? What happens then in a non-IPv6 environment?

Reply:
It won't behave any different than currently with [t/test01.t](https://github.com/zonemaster/zonemaster-backend/blob/develop/t/test01.t#L120) or [t/idn.t](https://github.com/zonemaster/zonemaster-backend/blob/develop/t/idn.t#L82).